### ISSUE TITLE
Lerna bootstrapping root only

### ISSUE THREAD

**Issue by vlopp** [lerna/lerna#2567]
**Under what circumstances does lerna bootstrap only the root package?**

Since a while, `lerna bootstrap` doesn't bootstrap all of the packages, and the below is logged:
```
lerna info bootstrap root only
```

* Moreover, reverting commits (even to those where everything worked) seems to have no impact, and Lerna still bootstraps root only.

* In new, simple Lerna projects, the same behviour occurs.

* Trying older Lerna versions did not help.

Unfortunately, the project's closed source.

```json
{
  "packages": ["packages/*"],
  "version": "0.0.0",
  "useWorkspaces": true,
  "npmClient": "yarn",
  "npmClientArgs": ["--ignore-engines"]
}

```

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.20.2 |
| `npm --version`  | 6.13.7 |
| `yarn --version` | 1.22.4 |
| `node --version` | v13.11.0 |

| OS | Version |
| Ubuntu | 19.10 |

**Comment by vlopp** [comment]
I must've been drunk that evening.

Everything works fine. Lots of packages suddenly seem to have their node_modules hoisted in their entirety, but it... works.

Feel free to delete this one.

**Comment by codemaster138** [comment]
Strange... I'm having exactly this problem; been having it for days; and I'm certain I'm *not* drunk

**Comment by ikeedge** [comment]
.

**Comment by Enrice** [comment]
after playing around with npm and yarn workspaces, I reverted everything - but lerna still refuses to operate on subpackages. only this exact message is displayed: "lerna info bootstrap root only"

**Comment by puranjayjain** [comment]
This issue still exists needs to be re-opened.

**Comment by eamon0989** [comment]
I am currently facing this issue, using Lerna `v5.5.1` and `npm` `v8.15.0`.

**Comment by JaneJeon** [comment]
@eamon0989 your fixed worked for me, but this is honestly such a huge footgun. Is this behaviour documented anywhere (i.e. `useWorkspaces` is only for yarn)?

**Comment by eamon0989** [comment]
@JaneJeon I had a quick look and I found it documented here:
https://github.com/lerna/lerna/blob/main/commands/bootstrap/README.md#--use-workspaces

> Enables integration with [Yarn Workspaces](https://github.com/yarnpkg/rfcs/blob/master/implemented/0000-workspaces-install-phase-1.md) (available since yarn@0.27+). The values in the array are the commands in which Lerna will delegate operation to Yarn (currently only bootstrapping). If --use-workspaces is true then packages will be overridden by the value from package.json/workspaces., and both --ignore and --scope will be ignored.

**Comment by ghiscoding** [comment]
> Edit: It turns out that I was getting this because I had `"useWorkspaces": true` set in `lerna.json`, but I am using `npm`, and `useWorkspaces` is for `yarn`.

I just want to point out, that it's no longer just specific to Yarn, npm also support "workspaces" in the package.json, see npm [defining workspaces](https://docs.npmjs.com/cli/v8/using-npm/workspaces#defining-workspaces)
